### PR TITLE
Adding ability to hide posts.

### LINKED ISSUE #1403: Ability to hide posts manually
### Describe the solution you'd like

As a user, I would like to have the ability to manually hide posts from showing up on my feed. This is the same functionality available on Reddit. 

### Describe alternatives you've considered

There is no way to do this AFAIK

There is another request for hiding posts but it's about automatically hiding read posts which is NOT the same as what I am requesting. https://github.com/LemmyNet/lemmy/issues/1072

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/96588646-ability-to-hide-posts-manually?utm_campaign=plugin&utm_content=tracker%2F126011972&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F126011972&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by dessalines**
- Adds an post/hide API route.
- Adds a `show_hidden` (default false) to `GetPosts`.
- Adds a `hidden` field to `PostView`.
- Removes the single `post_id` from MarkPostAsRead.
- Fixes #1403

### CODE REVIEW

File: crates/api_common/src/post.rs [hunk 1]
```diff
@@ -148,12 +149,20 @@ pub struct RemovePost {
 #[cfg_attr(feature = "full", ts(export))]
 /// Mark a post as read.
 pub struct MarkPostAsRead {
-  /// TODO: deprecated, send `post_ids` instead
```
Comment (dessalines) [inline crates/api_common/src/post.rs:151]: Did this cleanup task here.

File: crates/db_schema/src/impls/post.rs [hunk 2]
```diff
@@ -1,23 +1,10 @@
 use crate::{
   newtypes::{CommunityId, DbUrl, PersonId, PostId},
-  schema::post::dsl::{
```
Comment (dessalines) [inline crates/db_schema/src/impls/post.rs:3]: Removed these dsl imports as part of a cleanup.

File: crates/db_schema/src/impls/post.rs [hunk 3]
```diff
@@ -329,13 +311,48 @@ impl PostRead {
...
+        .filter(post_read::post_id.eq_any(post_id_))
+        .filter(post_read::person_id.eq(person_id_)),
+    )
+    .execute(conn)
+    .await
+  }
+}
+
+impl PostHide {
+  pub async fn hide(
```
Comment (dessalines) [inline crates/db_schema/src/impls/post.rs:327]: Main change is here, just adding hide and unhide functions that work the same as `post_read`

File: crates/db_views/src/post_view.rs [hunk 4]
```diff
@@ -402,7 +421,14 @@ fn queries<'a>() -> Queries<
       // Do not hide read posts when it is a user profile view
       // Or, only hide read posts on non-profile views
       if let (None, Some(person_id)) = (options.creator_id, my_person_id) {
-        query = query.filter(not(is_read(person_id)));
+        query = query.filter(not(is_read(person_id)))
```
Comment (Nutomic) [inline crates/db_views/src/post_view.rs:424]: This change seems wrong?

**Review by Nutomic — COMMENTED** [review]
Looks fine except for that one comment.

**Pull request merged**